File: platforms/ios/Dezel/Dezel/Core/JavaScriptPropertyNumberValue.swift
/**
 * @class JavaScriptPropertyNumberValue
 * @super JavaScriptPropertyValue
 * @since 0.7.0
 * @hidden
 */
open class JavaScriptPropertyNumberValue: JavaScriptPropertyValue {

	//--------------------------------------------------------------------------
	// MARK: Properties
	//--------------------------------------------------------------------------

	/**
	 * @property data
	 * @since 0.7.0
	 * @hidden
	 */
	private var data: Double

	//--------------------------------------------------------------------------
	// MARK: Methods
	//--------------------------------------------------------------------------

	/**
	 * @constructor
	 * @since 0.7.0
	 * @hidden
	 */
	public init(value: Double) {
		self.data = value
		super.init(type: .number)
	}

	/**
	 * @constructor
	 * @since 0.7.0
	 * @hidden
	 */
	public init(value: Double, unit: JavaScriptPropertyUnit) {

		self.data = value

		super.init(
			type: .number, unit: unit)
	}

	/**
	 * @method toString
	 * @since 0.7.0
	 */
	override open func toString() -> String {

		var string = self.data.toString()

		switch (self.unit) {

			case .px:
				string.append("px")
			case .pc:
				string.append("%")
			case .vw:
				string.append("vw")
			case .vh:
				string.append("vh")
			case .pw:
				string.append("pw")
			case .ph:
				string.append("ph")
			case .cw:
				string.append("cw")
			case .ch:
				string.append("ch")
			case .deg:
				string.append("deg")
			case .rad:
				string.append("rad")

			default:
				break
		}

		return string
	}

	/**
	 * @method toNumber
	 * @since 0.7.0
	 */
	override open func toNumber() -> Double {
		return self.data
	}

	/**
	 * @method toBoolean
	 * @since 0.7.0
	 */
	override open func toBoolean() -> Bool {
		return self.data > 0
	}
}

File: platforms/android/Dezel/dezel/src/main/java/ca/logaritm/dezel/core/JavaScriptPropertyNumberValue.kt
package ca.logaritm.dezel.core

import ca.logaritm.dezel.extension.type.toStr

/**
 * @class JavaScriptPropertyNumberValue
 * @super JavaScriptPropertyValue
 * @since 0.7.0
 */
open class JavaScriptPropertyNumberValue(value: Double, unit: JavaScriptPropertyUnit = JavaScriptPropertyUnit.NONE): JavaScriptPropertyValue(JavaScriptPropertyType.NUMBER, unit) {

	//--------------------------------------------------------------------------
	// Properties
	//--------------------------------------------------------------------------

	/**
	 * @property value
	 * @since 0.7.0
	 * @hidden
	 */
	private var data: Double

	//--------------------------------------------------------------------------
	// Methods
	//--------------------------------------------------------------------------

	/**
	 * @constructor
	 * @since 0.7.0
	 */
	init {
		this.data = value
	}

	/**
	 * @method toString
	 * @since 0.7.0
	 */
	override fun toString(): String {

		val string = StringBuilder(this.data.toStr())

		when (this.unit) {
			JavaScriptPropertyUnit.PX  -> string.append("px")
			JavaScriptPropertyUnit.PC  -> string.append("%")
			JavaScriptPropertyUnit.VW  -> string.append("vw")
			JavaScriptPropertyUnit.VH  -> string.append("vh")
			JavaScriptPropertyUnit.PW  -> string.append("pw")
			JavaScriptPropertyUnit.PH  -> string.append("ph")
			JavaScriptPropertyUnit.CW  -> string.append("cw")
			JavaScriptPropertyUnit.CH  -> string.append("ch")
			JavaScriptPropertyUnit.DEG -> string.append("deg")
			JavaScriptPropertyUnit.RAD -> string.append("rad")
			else                       -> {}
		}

		return string.toString()
	}

	/**
	 * @method toNumber
	 * @since 0.7.0
	 */
	override fun toNumber(): Double {
		return this.data
	}

	/**
	 * @method toBoolean
	 * @since 0.7.0
	 */
	override fun toBoolean(): Boolean {
		return this.data > 0
	}
}